### PR TITLE
perf(js_analyze): use Box<str>/Box<[_]> to reduce memory usage

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

This PR replace `Vec<_>` and `String` with their boxed versions `Box<[_]>` and `Box<str>` in Rules' `Options` and `Signals`. This allows reducing memory usage and could improve perf in some scenario.

To do so I implemented `Deserializable` for `Box<[_]>` and `Box<str>`.

Also, I found some use of `text` instead of `trimmed_text`.
I took the opportunity of replacing `text` with `trimmed_text`.

## Test Plan

CI must pass.

### CODE REVIEW

**Comment by ematipico** [comment]
@Conaclos did you see concrete differences by any chance when running the CLI?

File: packages/@biomejs/biome/configuration_schema.json [hunk 1]
```diff
@@ -4132,7 +4132,6 @@
 			"properties": {
 				"attributes": {
 					"description": "Additional attributes that will be sorted.",
-					"default": ["class", "className"],
```
Comment (ematipico) [inline packages/@biomejs/biome/configuration_schema.json:4135]: Is it expected?

**Review by ematipico — APPROVED** [review]
Since you changed many lint rules, you might as well want to update the contribution guide of the analyzer, and add a chapter that talks about it.

File: packages/@biomejs/biome/configuration_schema.json (hunk 1 above)
Comment (Conaclos) [reply packages/@biomejs/biome/configuration_schema.json:4135]: Yes. This avoids some allocations and doesn't change the behavior of the rule.

**Pull request merged**